tests/integration/targets/web_application/tasks/tests.yml

```diff
@@ -0,0 +1,111 @@
+---
+- name: test site exists, but stopped in case of duplicate web binding
+  website:
+    name: "{{ test_site_name }}"
+    state: stopped
+    physical_path: 'C:\inetpub\wwwroot'
+
+- name: test app is absent (baseline)
+  web_application:
+    state: absent
+    site: "{{ test_site_name }}"
+    name: "{{ test_app_name }}"
+
+- name: create test app
+  web_application:
+    state: present
+    site: "{{ test_site_name }}"
+    name: "{{ test_app_name }}"
+    physical_path: "{{ test_physical_path }}"
+  register: result
+
+- name: Fetch info for test app
+  web_application_info:
+    name: "{{ test_app_name }}"
+  register: test_info
+
+- assert:
+    that:
+    - result.changed == true
+    - test_info.applications[0].physical_path == test_physical_path
+
+- name: create test app (idempotent)
+  web_application:
+    state: present
+    site: "{{ test_site_name }}"
+    name: "{{ test_app_name }}"
+    physical_path: "{{ test_physical_path }}"
+  register: result
+
+- assert:
+    that:
+    - result.changed == false
+
+- name: set test app credentials
+  web_application:
+    state: present
+    site: "{{ test_site_name }}"
+    name: "{{ test_app_name }}"
+    connect_as: specific_user
+    username: "{{ test_user }}"
+    password: "{{ test_password }}"
+  register: result
+
+- name: Fetch info for test app creds
+  web_application_info:
+    name: "{{ test_app_name }}"
+  register: test_info
+
+- assert:
+    that:
+    - result.changed == true
+    - test_info.applications[0].physical_path == test_physical_path
```

Comment:
Seems like we may want to retrieve the `username` in `web_application_info` so we can properly test that `connect_as: specific_user` worked or not.